File: app/src/main/java/com/iraq/nireader/data/model/ApiPayload.kt
package com.iraq.nireader.data.model

import kotlinx.serialization.Serializable

/**
 * Standard request payload for POST /api/national-id/read sent to Desktop Host.
 */
@Serializable
data class ApiCardReadRequest(
    val deviceId: String,
    val appVersion: String = "1.0.0",
    val readTimestamp: Long,
    val cardData: CardData
)

/**
 * Expected response from Desktop Server.
 */
@Serializable
data class ApiCardReadResponse(
    val status: ApiResponseStatus,
    val message: String? = null,
    val transactionId: String? = null
)

@Serializable
enum class ApiResponseStatus {
    SUCCESS,
    FAILED,
    INVALID_DATA,
    UNAUTHORIZED,
    SERVER_ERROR
}

File: flutter_ni_reader/lib/data/models/api_payload.dart
import 'card_data.dart';

enum ApiResponseStatus {
  SUCCESS,
  ERROR;

  String toJson() => name;
  static ApiResponseStatus fromJson(String value) =>
      ApiResponseStatus.values.firstWhere((e) => e.name == value, orElse: () => ApiResponseStatus.ERROR);
}

class ApiCardReadRequest {
  final String deviceId;
  final int readTimestamp;
  final CardData cardData;

  ApiCardReadRequest({
    required this.deviceId,
    required this.readTimestamp,
    required this.cardData,
  });

  Map<String, dynamic> toJson() => {
    'deviceId': deviceId,
    'readTimestamp': readTimestamp,
    'cardData': cardData.toJson(),
  };
}

class ApiCardReadResponse {
  final ApiResponseStatus status;
  final String message;
  final String? cardId;

  ApiCardReadResponse({
    required this.status,
    required this.message,
    this.cardId,
  });

  factory ApiCardReadResponse.fromJson(Map<String, dynamic> json) => ApiCardReadResponse(
    status: ApiResponseStatus.fromJson(json['status'] ?? 'SUCCESS'),
    message: json['message'] ?? '',
    cardId: json['cardId'],
  );
}
